src/installer/__main__.py

```diff
@@ -0,0 +1,78 @@
+"""Installer CLI."""
+
+import argparse
+import os.path
+import sys
+import sysconfig
+from typing import Dict, Optional, Sequence
+
+import installer
+import installer.destinations
+import installer.sources
+import installer.utils
+
+
+def main_parser() -> argparse.ArgumentParser:
+    """Construct the main parser."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("wheel", type=str, help="wheel file to install")
+    parser.add_argument(
+        "--destdir",
+        "-d",
+        metavar="path",
+        type=str,
+        help="destination directory (prefix to prepend to each file)",
+    )
+    parser.add_argument(
+        "--compile-bytecode",
+        action="append",
+        metavar="level",
+        type=int,
+        choices=[0, 1, 2],
+        help="generate bytecode for the specified optimization level(s) (default=0, 1)",
+    )
+    parser.add_argument(
+        "--no-compile-bytecode",
+        action="store_true",
+        help="don't generate bytecode for installed modules",
+    )
+    return parser
+
+
+def get_scheme_dict(distribution_name: str) -> Dict[str, str]:
+    """Calculate the scheme dictionary for the current Python environment."""
+    scheme_dict = sysconfig.get_paths()
+
+    # calculate 'headers' path, not currently in sysconfig -
+    # see https://bugs.python.org/issue44445. This copies what distutils does.
+    scheme_dict["headers"] = os.path.join(scheme_dict["include"], distribution_name)
```

Comment:
This doesn't actually copy what distutils does because (a) `include` is always on the `sys.base_prefix`, meaning that  installer will attempt to install headers under the base prefix, and (b) when the root is "platlib", the installation will be performed under "purelib".

Reply:
I see what you mean about a) - distutils replaces `installed_base` (= `sys.base_prefix`) with `base` (= `sys.prefix`). We can do the equivalent easily enough, I think.

I'm not sure what you mean about b) though. I don't think that root-is-purelib setting should make any difference to where the headers are installed, and I haven't spotted anything in distutils referring to another destination for them (though I could easily have overlooked it, the logic is fairly spread out).

Reply:
I assume that if the root is platlib then the headers should be installed in platinclude, i.e. under `sys.(base_)exec_prefix`.

Reply:
distutils doesn't do this, never mind.

Reply:
I don't see distutils implementing that, and it doesn't seem right to me. The wheel contents are 5 separate components to be installed - purelib and platlib are separate, but there's just one `headers` directory, no `platheaders`. As a convenience, the root of the wheel is considered either platlib or purelib, but I don't think that should affect where any of the other components go.

Reply:
OK, great. I'll deal with prefix vs base_prefix at some point later.

Reply:
The latest change - 275e73daf89de4fde54fcfc2ec187ecbdda22cd4 - should use prefix in place of base_prefix so headers can be installed in a virtualenv.